asserts/header_checks.go

```diff
@@ -284,13 +284,17 @@ func checkStringMatchesWhat(headers map[string]interface{}, name, what string, p
 }
 
 func checkOptionalBool(headers map[string]interface{}, name string) (bool, error) {
+	return checkOptionalBoolWhat(headers, name, "header")
+}
+
+func checkOptionalBoolWhat(headers map[string]interface{}, name, what string) (bool, error) {
```

Comment:
nitpick: I'd rename these two

- checkOptionalBool -> checkOptionalBoolHeader
- checkOptionalBoolWhat -> checkOptionalBool

Reply:
Though maybe, after a second look, "header" should always be written, and the `what` would be an additional context. So, `checkOptionalBool` could stay as it is, and `checkOptionalBoolWhat` be renamed to `checkOptionalBoolWithContext` and its format string be changed to
```go
"%s: %q header must be 'true' or 'false'", context, name)
```
But maybe it's for a follow-up, it's not critical.

Reply:
we have a bunch of helpers like these already, if we change the names etc we need to change all of them so follow-up material